### PR TITLE
fix(example): mistakes in fuzz_x8664_linux binary

### PR DESCRIPTION
**Pull request by chinggg**
### Which kind of PR do you create?

- [x] This PR only contains minor fixes.
- [ ] This PR contains major feature update.
- [ ] This PR introduces a new function/api for Qiling Framework.

### Coding convention?

- [x] The new code conforms to Qiling Framework naming convention.
- [x] The imports are arranged properly.
- [ ] Essential comments are added.
- [ ] The reference of the new code is pointed out.

### Extra tests?

- [ ] No extra tests are needed for this PR.
- [ ] I have added enough tests for this PR.
- [ ] Tests will be added after some discussion and review.

### Changelog?

- [x] This PR doesn't need to update Changelog.
- [ ] Changelog will be updated after some proper review.
- [ ] Changelog has been updated in my PR.

### Target branch?

- [x] The target branch is dev branch.

### One last thing

- [x] I have read the [contribution guide](https://docs.qiling.io/en/latest/contribution/)

-----

IMO, `fuzz.c` intended to read from `stdin` to buf and copy it to buf2, then print buf2 as string.
But there are two mistakes which do not affect the fuzzing result, I think they are typos:
- `*buf = getc(stdin) == 'A'` does not reads char, but the result of comparison
- `printf(buf2)` does not print buf2 as string

So I fix `fuzz.c` then compile it with `gcc -g` on Ubuntu 20.04 + gcc9.3, and modify the according address of `main` and `stack_chk` in `fuzz_x8664_linux.py`, it works as before.

If you think the fix is correct, the `arm_qnx` example may need fix as well.

### CODE REVIEW

File: examples/fuzzing/linux_x8664/fuzz.c [hunk 1]
```diff
@@ -10,13 +10,13 @@ int fun(int i)
...
 
-    while (*buf = getc(stdin) == 'A')
+    while ((*buf = getc(stdin)) == 'A')
     {
         buf[i++] = *buf;
     }
 
     strncpy(buf2, buf, i);
-    printf(buf2);
+    printf("%s", buf2);
```
Comment (elicn) [inline examples/fuzzing/linux_x8664/fuzz.c:19]: This is actually fixing an _Uncontrolled Format String_ vulnerability that could be triggered if `buf` was not limited only to `'A'` chars. Anyway, consider changing that to `puts(buf2)` that is functionally identical but much more efficient where there is no format string involved.

**Review by elicn — APPROVED** [review]
Good catch!

File: examples/fuzzing/linux_x8664/fuzz.c (hunk 1 above)
Comment (chinggg) [reply examples/fuzzing/linux_x8664/fuzz.c]: Thanks! This has been changed.

**Comment by xwings** [comment]
Hey, thanks and welcome to Qiling !

**Pull request merged**